.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,754 @@
+            '::set-output name=is-stable-abi::{is_stable_abi}'.
+            format(is_stable_abi=str(is_stable_abi).lower())
+        )
+      shell: python
+
+    - name: >-
+        Calculate Python interpreter version hash value
+        for use in the cache key
+      if: fromJSON(steps.py-abi.outputs.is-stable-abi)
+      id: calc-cache-key-py
+      run: |
+        from hashlib import sha512
+        from sys import version
+        hash = sha512(version.encode()).hexdigest()
+        print('::set-output name=py-hash-key::{hash}'.format(hash=hash))
+      shell: python
+
+    - name: Get pip cache dir
+      if: fromJSON(steps.py-abi.outputs.is-stable-abi)
+      id: pip-cache-dir
+      run: >-
+        echo "::set-output name=dir::$(python -m pip cache dir)"
+
+    - name: Set up pip cache
+      if: fromJSON(steps.py-abi.outputs.is-stable-abi)
+      uses: actions/cache@v2.1.7
+      with:
+        path: ${{ steps.pip-cache-dir.outputs.dir }}
+        key: >-
+          ${{ runner.os }}-pip-${{
+          steps.calc-cache-key-py.outputs.py-hash-key }}-${{
+          needs.pre-setup.outputs.cache-key-files }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{
+              steps.calc-cache-key-py.outputs.py-hash-key
+          }}-
+          ${{ runner.os }}-pip-
+
+    - name: Update pip
+      run: >-
+        python -m
+        pip install
+        --user
+        pip
+
+    - name: Grab the source from Git
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.inputs.release-commitish }}
+
+    - name: Download all the dists
+      uses: actions/download-artifact@v2
+      with:
+        name: python-package-distributions
+        path: dist/
+
+    - name: Install dependencies
+      run: >-
+        python -m
+        pip install
+        --user
+        --requirement requirements-dev.txt
+
+    - name: Install previously built wheel
+      run: >-
+        python -m
+        pip install
+        --no-cache-dir
+        --user
+        --no-index
+        'dist/${{ needs.pre-setup.outputs.wheel-artifact-name }}'
+
+    - name: >-
+        Log platform.platform()
+      run: >-
+        python -m platform
+    - name: >-
+        Log platform.version()
+      run: >-
+        python -c "import platform;
+        print(platform.version())"
+    - name: >-
+        Log platform.uname()
+      run: >-
+        python -c "import platform;
+        print(platform.uname())"
+    - name: >-
+        Log platform.release()
+      run: >-
+        python -c "import platform;
+        print(platform.release())"
+    - name: Log stdlib OpenSSL version
+      run: >-
+        python -c
+        "import ssl; print('\nOPENSSL_VERSION: '
+        + ssl.OPENSSL_VERSION + '\nOPENSSL_VERSION_INFO: '
+        + repr(ssl.OPENSSL_VERSION_INFO)
+        + '\nOPENSSL_VERSION_NUMBER: '
+        + repr(ssl.OPENSSL_VERSION_NUMBER))"
+
+    - name: Check rst
+      run: |
+        python setup.py check --restructuredtext
```

Comment:
With the suggestion to drop `setup.py`, I suggest replacing the linter that checks RST. After all, invoking `setup.py` is actively discouraged: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html.

Reply:
that was migrated from the current implementation, should probably get replaced by `twine check`, but i didn't want to include that in this PR

Reply:
replaced by `twine check --strict` in #746